SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -751,6 +955,99 @@ bool FSpatialGDKEditorToolbarModule::StopSpatialServiceIsVisible() const
 	return SpatialGDKSettings->bShowSpatialServiceButton && LocalDeploymentManager->IsSpatialServiceRunning();
 }
 
+void FSpatialGDKEditorToolbarModule::OnToggleSpatialNetworking()
+{
+	UGeneralProjectSettings* GeneralProjectSettings = GetMutableDefault<UGeneralProjectSettings>();
+	UProperty* SpatialNetworkingProperty = UGeneralProjectSettings::StaticClass()->FindPropertyByName(FName("bSpatialNetworking"));
+
+	GeneralProjectSettings->SetUsesSpatialNetworking(!GeneralProjectSettings->UsesSpatialNetworking());
+	GeneralProjectSettings->UpdateSinglePropertyInConfigFile(SpatialNetworkingProperty, GeneralProjectSettings->GetDefaultConfigFilename());
+}
+
+bool FSpatialGDKEditorToolbarModule::OnIsSpatialNetworkingEnabled() const
+{
+	return GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking();
+}
+
+void FSpatialGDKEditorToolbarModule::GDKEditorSettingsClicked() const
+{
+	FModuleManager::LoadModuleChecked<ISettingsModule>("Settings").ShowViewer("Project", "SpatialGDKEditor", "Editor Settings");
+}
+
+void FSpatialGDKEditorToolbarModule::GDKRuntimeSettingsClicked() const
+{
+	FModuleManager::LoadModuleChecked<ISettingsModule>("Settings").ShowViewer("Project", "SpatialGDKEditor", "Runtime Settings");
+}
+
+bool FSpatialGDKEditorToolbarModule::IsNoAutomaticConnectionSelected() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	return SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::NoAutomaticConnection;
+}
+
+bool FSpatialGDKEditorToolbarModule::IsLocalDeploymentSelected() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	return SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::LocalDeployment;
+}
+
+bool FSpatialGDKEditorToolbarModule::IsCloudDeploymentSelected() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
+	return SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::CloudDeployment;
+}
+
+bool FSpatialGDKEditorToolbarModule::IsSpatialOSNetFlowConfigurable() const
+{
+	return OnIsSpatialNetworkingEnabled() && !(LocalDeploymentManager->IsLocalDeploymentRunning());
+}
+
+void FSpatialGDKEditorToolbarModule::NoAutomaticConnectionClicked()
+{
+	USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetMutableDefault<USpatialGDKEditorSettings>();
+	SpatialGDKEditorSettings->SpatialOSNetFlowType = ESpatialOSNetFlow::NoAutomaticConnection;
```

Comment:
Setter on  SpatialGDKEditorSettings for SpatialOSNetFlowType  might be useful here

Reply:
Hmm, this doesn't quite work for the cloud connection flow, because we generate the dev auth token asynchronously and only set the runtime use development authentication flow if and when that succeeds.